python/pyspark/sql/functions.py

```diff
@@ -33,17 +33,36 @@
 __all__ = ['countDistinct', 'approxCountDistinct', 'udf']
 
 
-def _create_function(name, doc=""):
+def _function_obj(sc, is_math=False):
+    if not is_math:
+        return sc._jvm.functions
+    else:
+        return sc._jvm.mathfunctions
+
+
+def _create_function(name, doc="", is_math=False, binary=False):
```

Comment:
I did... The problem is, I can't pass in `sc` when it was outside this function. I guess the problem was globally calling `sc = SparkContext._active_spark_context`, meaning outside of a function. That's why I had to have this hack.